gax-java/gax/src/main/java/com/google/api/gax/rpc/Callables.java

```diff
@@ -51,6 +51,16 @@ public class Callables {
 
   private Callables() {}
 
+  public static <RequestT, ResponseT> UnaryCallable<RequestT, ResponseT> retrying(
+      UnaryCallable<RequestT, ResponseT> innerCallable,
+      RetryAlgorithm<ResponseT> retryAlgorithm,
+      ClientContext clientContext) {
+    ScheduledRetryingExecutor<ResponseT> retryingExecutor =
+        new ScheduledRetryingExecutor<>(retryAlgorithm, clientContext.getExecutor());
+    return new RetryingCallable<>(
+        clientContext.getDefaultCallContext(), innerCallable, retryingExecutor);
+  }
```

Comment:
Do you have an example of how BigTable is planning on using this? IIUC, most of the methods here are used by the TransportCallableFactory to build the callables and I believe the CallableFactory is invoked from the TransportServiceStub (which is generated).

Reply:
I staged the example here: https://github.com/googleapis/java-bigtable/pull/2055/files#diff-66bb41fee1ea55e7ab391ed8827b46e57035b81982c587251d577028866fade0, in the EnhancedBigtableStub class.

Reply:
Thanks @mutianf, for tracking purposes, do you mind creating an issue and link the example above to the issue?

Reply:
> Is this change b/c BigTable wants to set custom retry values that are different from what's in the grpc service config file and you don't want customers to set their own?

I'll give it a shot but Mattie can probably provide more info. In short, yes they want to set retry values from the server responses dynamically instead of a fixed value from service configs. See [internal docs](https://docs.google.com/document/d/1wo0KaD5FMrknOoyjSxPD_NGQ2NNJMb0WPVO9SzhwYCU/edit?tab=t.0#heading=h.x9snb54sjlu9) for details.

Reply:
> Is this change b/c BigTable wants to set custom retry values that are different from what's in the grpc service config file and you don't want customers to set their own?

We want to use a different RetryAlgorithm to look at the response RetryInfo: https://github.com/googleapis/java-bigtable/blob/main/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/gaxx/retrying/RetryInfoRetryAlgorithm.java. There's another internal doc that's focused on this use case: https://docs.google.com/document/d/1Kim-80vUSDgYxDVrFtkl8NkV-SsW8KAor_Ca1rpvbyo/edit?resourcekey=0-oV2CmU7_sZEMpPMFF9ZSOA&tab=t.0#heading=h.j0ev05qqzdu4.


@blakeli0  sure I'll create an issue.

Reply:
Thanks Mattie! I'll read through the doc for more information regarding this. I have a better idea of the use case and just need to fully confirm my understanding.